src/hotspot/share/gc/serial/cardTableRS.cpp

```diff
@@ -54,9 +54,8 @@ void CardTableRS::verify_used_region_at_save_marks(Space* sp) const {
 }
 #endif
 
-void CardTableRS::maintain_old_to_young_invariant(Generation* old_gen, bool is_young_gen_empty) {
-  assert(SerialHeap::heap()->is_old_gen(old_gen), "precondition");
-
+void CardTableRS::maintain_old_to_young_invariant(TenuredGeneration* old_gen,
+                                                  bool is_young_gen_empty) {
```

Comment:
The cpp file needs to include `tenuredGeneration.hpp` now. (`generation.hpp` is still required).